web/info/src/stores/theme.ts

```diff
@@ -1,5 +1,41 @@
 import { writable } from 'svelte/store';
+import { useMediaQuery } from 'svelte-breakpoints';
 
-import { APP_THEMES } from '$lib/constants';
+import { browser } from '$app/environment';
+import { APP_THEME_COOKIE_NAME, APP_THEMES } from '$lib/constants';
 
-export default writable<(typeof APP_THEMES)[keyof typeof APP_THEMES]>(APP_THEMES.light);
+const initialValue = browser
+  ? (document.cookie
+      ?.split('; ')
+      ?.find((row) => row.startsWith(APP_THEME_COOKIE_NAME))
+      ?.split('=')[1] as typeof APP_THEMES[number]) ?? APP_THEMES[0]
+  : APP_THEMES[0];
+
+const theme = writable<typeof APP_THEMES[0 | 1]>(initialValue);
+
+const prefersDark = useMediaQuery('(prefers-color-scheme: dark)'),
+  prefersLight = useMediaQuery('(prefers-color-scheme: light)');
+
+if (browser) {
+  // Only set theme automatically if no cookie is set
```

Comment:
That's true, but the alternative is to handle saving the cookie from the button and not on updates of the Store, which I don't *love* but I suppose it would achieve that...

Reply:
Hm, is there a way to provide the default value for the store? In Nuxt, you'd do `useState('theme', is dark() ? 'dark' :'light')` and only updates to the theme would trigger setting the cookie.

Reply:
The default is whatever it's instantiated with - but that only happens client-side